pssh/ssh_client.py

```diff
@@ -100,6 +100,14 @@ def __init__(self, host,
                             else host)
         _user = host_config['user'] if 'user' in host_config else None
         user = user if user else _user
+
+        _port = int(host_config['port']) if 'port' in host_config else 22
+        port = port if port else _port
+
+        # FIXME: we take the first key
+        _pkey = paramiko.RSAKey.from_private_key_file(host_config['identityfile'][0]) if 'identityfile' in host_config else None
```

Comment:
Yup, either works for me. Probably simpler to try/except each type. The exception raised is `paramiko.SSHException` for invalid key type file load.

Looking at paramiko's source, it checks header matches key type before trying to load the key already so we would be duplicating that code.

Agree that paramiko would ideally have a convenience function for guessing key type, might be worth a PR if either of us find the time :)